standard_sequence/experiment_components.py

```diff
@@ -0,0 +1,816 @@
+        self.start_tweezers(t)
+
+    def start_tweezers(self, t):
+        assert shot_globals.do_sequence_mode, "shot_globals.do_sequence_mode is False, running Fifo mode now. Set to True for sequence mode"
+        spectrum_manager.start_card()
+        t1 = spectrum_manager.start_tweezers(t)
+        print("tweezer start time:", t1)
+        self.aom_on(t, self.tw_power)
+
+    def stop_tweezers(self, t):
+        # stop tweezers
+        t2 = spectrum_manager.stop_tweezers(t)
+        print("tweezer stop time:", t2)
+
+        # TODO: explain what this does, Answer: only when there is a dummy segment the spectrum card will actually output waveform, a bug related to the spectrum card server
+
+        # dummy segment
+        t1 = spectrum_manager.start_tweezers(t)
+        print("tweezer start time:", t1)
+        t += 2e-3
+        t2 = spectrum_manager.stop_tweezers(t)
+        print("tweezer stop time:", t2)
+        spectrum_manager.stop_card(t)
+        return t
+
+    def intensity_servo_keep_on(self, t):
+        """keep the AOM digital high for intensity servo"""
+        self.aom_on(t, 1)
+
+    def aom_on(self, t, const):
+        """Turn on the tweezer beam using aom"""
+        devices.tweezer_aom_digital.go_high(t)  # digital on
+        devices.tweezer_aom_analog.constant(t, const)  # analog on
+        self.tw_power = const
+
+    def aom_off(self, t):
+        """Turn off the tweezer beam using aom"""
+        devices.tweezer_aom_digital.go_low(t)  # digital off
+        devices.tweezer_aom_analog.constant(t, 0)  # analog off
+        self.tw_power = 0
+
+    def ramp_power(self, t, dur, final_power):
+        devices.tweezer_aom_analog.ramp(
+            t, duration=dur, initial=self.tw_power, final=final_power, samplerate=1e5
+        )
+        self.tw_power = final_power
+        return t + dur
+
+    def sine_mod_power(self, t, dur, amp, freq):
+        devices.tweezer_aom_analog.sine(
+            t,
+            duration=dur,
+            amplitude=amp,
+            angfreq=2 * np.pi * freq,
+            phase=0,
+            dc_offset=self.tw_power,
+            samplerate=1e5,
+        )
+
+
+class Microwave:
+    def __init__(self, t):
+        CONST_SPECTRUM_UWAVE_CABLE_ATTEN = (
+            4.4  # cable attenutation, dB  meausred at 300 MHz
+        )
+        self.mw_detuning = shot_globals.mw_detuning  # tune the microwave detuning here
+        self.uwave_dds_switch_on = True
+        self.uwave_absorp_switch_on = False
+        self.spectrum_uwave_power = -1  # dBm power set at the input of dds switch, this power is set to below the amplifier damage threshold
+        devices.uwave_dds_switch.go_high(
+            t
+        )  # dds_switch always on, can be off if need further higher distinguishing ratio
+        devices.uwave_absorp_switch.go_low(
+            t
+        )  # absorp switch only on when sending pulse
+        devices.spectrum_uwave.set_mode(
+            replay_mode=b"sequence",

```

Comment:
"b" is a typo?

Reply:
No, this is not a typo. This is python Byte Strings usage.